File: 0029-divide-two-integers/0029-divide-two-integers.py
class Solution:
    def divide(self, dividend: int, divisor: int) -> int:
        quotient = int(dividend/divisor)
        if quotient>=2**31-1:
            return 2**31-1
        elif quotient<=pow(-2,31):
            return pow(-2,31)
        return quotient

File: 0029-divide-two-integers/0029-divide-two-integers.js
/**
 * @param {number} dividend
 * @param {number} divisor
 * @return {number}
 */
var divide = function(dividend, divisor) {
    let quotient = Math.trunc(dividend/divisor)
    if (quotient>=(2**31 - 1)) return 2**31-1
    else if (quotient<= (Math.pow(-2,31))) return Math.pow(-2,31)
    else return quotient
    
};